docs/book/validator.md

```diff
@@ -33,3 +34,31 @@ $validator->isValid('myIdentity', [
      'myCredentialContext' => 'myCredential',
 ]);
 ```
+
+## Configuring custom authentication result codes and messages
+
+Constructor configuration option `code_map` is a map of custom authentication result
+codes to validation messages keys.
+
+`code_map` can specify custom validation message key. New message template
+will be registered for that key, which can further be customized
+using `Validator::setMessage()` method or `messages` configuration option.
```

Comment:
Let's try the following instead:

```markdown
A `code_map` value can map to a custom validation message key if desired. If you do, you should also register a message template for that key via either the `messages` configuration option or the `AuthenticationValidator::setMessage()` method:
```

Reply:
Setting message template is not required, it defaults to template for `general` error.

Reply:
So make that explicit in the documentation, so somebody not familiar with how validators work has everything they need to understand it.

Reply:
`AbstractValidator::setMessage()` can only be invoked for existing message key provided by validator
```php
        if (! isset($this->abstractOptions['messageTemplates'][$messageKey])) {
            throw new Exception\InvalidArgumentException("No message template exists for key '$messageKey'");
        }
```

What custom message key in `code_map` does is adds new entry to `messageTemplates`. 
After that `messages` option or `setMessage()` can be used with that key